tests/bindings/test_measurements_nb.py

```diff
@@ -0,0 +1,403 @@
+        meas = MeasurementsClass(sv)
+        probs = meas.probs([0])
+
+        # Should be [1.0, 0.0]
+        assert np.allclose(probs, [1.0, 0.0], atol=1e-6)
+
+        # Apply Hadamard
+        sv.Hadamard([0], False, [])
+
+        # Get the state
+        result = np.zeros(2, dtype=dtype)
+        sv.getState(result)
+
+        # Check probabilities
+        meas = MeasurementsClass(sv)
+        probs = meas.probs([0])
+
+        # Should be [0.5, 0.5]
+        assert np.allclose(probs, [0.5, 0.5], atol=1e-6)
+
+    @pytest.mark.parametrize("module_name", modules)
+    def test_measurements_generate_samples(
+        self, module_name, get_classes, precision, nanobind_module
+    ):
+        """Test sample generation."""
+        # Extract backend name from module_name
+        backend_name = module_name.split(".")[-1].split("_")[1]
+        module = nanobind_module(backend_name)
+
+        StateVectorClass, MeasurementsClass, _, _ = get_classes(module, precision)
+
+        num_qubits = 2
+        # Create a numpy array representing |0> state with appropriate size
+        dtype = np.complex64 if precision == "64" else np.complex128
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[0] = 1.0
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Create Bell state
+        sv.Hadamard([0], False, [])
+        sv.CNOT([0, 1], False, [])
+
+        meas = MeasurementsClass(sv)
+
+        # Generate samples
+        num_samples = 1000
+        samples = meas.generate_samples(num_qubits, num_samples)
+
+        # Check shape
+        assert samples.shape == (num_samples, 2)
+
+        # In a Bell state, the qubits should be perfectly correlated
+        # Count how many times we get |00⟩ and |11⟩
+        count_00 = np.sum(np.all(samples == [0, 0], axis=1))
+        count_11 = np.sum(np.all(samples == [1, 1], axis=1))
+
+        # We should have approximately 50% |00⟩ and 50% |11⟩
+        # and very few or no |01⟩ and |10⟩
+        assert count_00 + count_11 >= 0.95 * num_samples  # Allow for small statistical fluctuations
+        assert 0.4 <= count_00 / num_samples <= 0.6  # Allow for statistical fluctuations
+        assert 0.4 <= count_11 / num_samples <= 0.6  # Allow for statistical fluctuations
+
+        # Test generate_samples for all wires
+        all_samples = meas.generate_samples(num_qubits, num_samples)
+        assert all_samples.shape == (num_samples, num_qubits)
+
+        # The results should be the same as when specifying all wires
+        count_00_all = np.sum(np.all(all_samples == [0, 0], axis=1))
+        count_11_all = np.sum(np.all(all_samples == [1, 1], axis=1))
+
+        # The counts should be similar to the previous test
+        assert abs(count_00 - count_00_all) < 0.1 * num_samples
+        assert abs(count_11 - count_11_all) < 0.1 * num_samples
```

Comment:
```suggestion
        assert 0.4 * num_samples <= count_00_all <= 0.6 * num_samples
        assert 0.4 * num_samples <= count_11_all <= 0.6 * num_samples
```

These counts shouldn't depend directly on the outcome of the prior test. (e.g. what if the prior test had 40% |00> and this test has 60% |00>?)

Reply:
Thanks